src/management-system-v2/app/(dashboard)/[environmentId]/engines/engines-list.tsx

```diff
@@ -0,0 +1,237 @@
+          </Tooltip>
+        );
+      },
+      responsive: ['xs', 'sm'],
+    },
+    {
+      title: 'ID',
+      key: 'ID',
+      sorter: (a, b) => (a.id ? a.id.localeCompare(b.id || '') : -1),
+      render: (_, record) => <span>{record.id}</span>,
+      responsive: ['md'],
+    },
+  ];
+
+  if (isDiscoveredMode) {
+    columns.push({
+      title: 'Discovery Technology',
+      dataIndex: 'discoveryTechnology',
+      key: 'Discovery Technology',
+      sorter: (a, b) =>
+        (a as DiscoveredEngine).discoveryTechnology.localeCompare(
+          (b as DiscoveredEngine).discoveryTechnology,
+        ),
+      render: (_, record) => <span>{(record as DiscoveredEngine).discoveryTechnology}</span>,
+      responsive: ['md'],
+    });
+  } else {
+    columns.push({
+      title: 'Own Name',
+      dataIndex: 'ownName',
+      key: 'Own Name',
+      sorter: (a, b) => (a.ownName ? a.ownName.localeCompare(b.ownName || '') : -1),
+      render: (_, record) => <span>{record.ownName}</span>,
+      responsive: ['md'],
+    });
+  }
+
+  columns.push({
+    title: 'Description',
+    dataIndex: 'description',
+    key: 'Description',
+    sorter: (a, b) => (a.description ? a.description.localeCompare(b.description || '') : -1),
+    render: (_, record) => <span>{record.description}</span>,
+    responsive: ['md'],
+  });
+
+  const mobileColumnKeys = ['Name', 'Address', 'Description', 'Meta Data Button'];
+  const mobileColumns: TableColumnsType<Engine> = columns.filter((c) =>
+    mobileColumnKeys.includes(c.key as string),
+  );
+
+  mobileColumns.push({
+    fixed: 'right',
+    width: 100,
+    dataIndex: 'id',
+    key: 'Meta Data Button',
+    title: '',
+    render: (_, record) => {
+      return isDiscoveredMode ? (
+        <Button
+          style={{ float: 'right' }}
+          type="text"
+          onClick={() => onAction(ActionType.SAVE, record.id)}
+        >
+          <FaRegSave></FaRegSave>
+        </Button>
+      ) : (
+        <div style={{ float: 'right', display: 'flex', flexDirection: 'row' }}>
+          <Button type="text" onClick={() => onAction(ActionType.EDIT, record.id)}>
+            <EditOutlined />
+          </Button>
+          <Button type="text" onClick={() => onAction(ActionType.DELETE, record.id)}>
+            <DeleteOutlined />
+          </Button>
+        </div>
+      );
+    },
+    responsive: ['xs', 'sm'],
+  });
+
+  const allColumnTitles = ['Name', 'Hostname', 'ID', 'Address', 'Description'];
+
+  if (isDiscoveredMode) {
+    allColumnTitles.push('Discovery Technology');
+  } else {
+    allColumnTitles.push('Own Name');
+  }
+
+  const [selectedColumns, setSelectedColumns] = useState(columns);
+
+  return (
+    <>
+      <ElementList
+        tableProps={{
+          title: () => <Text type="secondary">{title}</Text>,
+          pagination: { pageSize: 10, position: ['bottomCenter'] },
+        }}
+        data={engines}
+        columns={breakpoint.xl ? selectedColumns : mobileColumns}
```

Comment:
There is no need to have to separate lists, you can just set the `responsive` property on each column accordingly.